useragent/user_agent_test.go

```diff
@@ -33,12 +33,16 @@ func TestFromContext_Default(t *testing.T) {
 func TestFromContext_Custom(t *testing.T) {
 	WithProduct("unit-tests", "0.0.1")
 	WithUserAgentExtra("pulumi", "3.8.4")
+	WithUserAgentExtra("terraform", "1.3.4")
+	WithUserAgentExtra("terraform", "1.3.5")
+	WithUserAgentExtra("terraform", "1.3.6")
 	ctx := InContext(context.Background(), "a", "b")
-	ctx = InContext(ctx, "terraform", "1.2.5")
+	ctx = InContext(ctx, "a", "d")
+	ctx = InContext(ctx, "a", "e")
```

Comment:
Opportunity to have ctx1 and ctx2 and test that it doesn't leak across contexts to test the copy.

Reply:
one sec, will add

Reply:
added